src/intershipCategories/internshipCategories.controllers.ts

```diff
@@ -0,0 +1,41 @@
+import { Body, Controller, Get, Param, Post, ParseUUIDPipe } from '@nestjs/common';
+import { InternshipCategoryService } from '../intershipCategories/internshipCategory.service';
+import { InternshipCategory } from '../intershipCategories/intershipCategory.entity';
+import { InternshipCategoryDto } from '../intershipCategories/dto/internshipCategory.dto';
+
+@Controller('/intcat')
+export class InternshipCategoryContoller {
+    constructor(
+        private readonly internshipcategoryService: InternshipCategoryService,
+
+    ){}
+
+    @Get()
+    async allCategories(): Promise<InternshipCategory[]> {
+      const internshipCategories = await this.internshipcategoryService.findAll()
+      return internshipCategories
+    }
+    @Post('')
+    async addIntCatergory(@Body() internshipCategory: InternshipCategoryDto) {
+      const newCategory = await this.internshipcategoryService.addCategory(internshipCategory)
+      return newCategory
+  
+    }
+  
+    @Get('/id/:id')
```

Comment:
/id/:id. what is this for ?????? @SunmadeBabafemi

Reply:
i created an endpoint to get a category via the id in the params. i will remove it as there is already an endpoint that gets a category via the category name in the request body